internal/services/azurestackhci/stack_hci_deployment_setting_resource.go

```diff
@@ -0,0 +1,1328 @@
+									Type:     pluginsdk.TypeBool,
+									Required: true,
+									ForceNew: true,
+								},
+
+								"credential_guard_enabled": {
+									Type:     pluginsdk.TypeBool,
+									Required: true,
+									ForceNew: true,
+								},
+
+								"drift_control_enabled": {
+									Type:     pluginsdk.TypeBool,
+									Required: true,
+									ForceNew: true,
+								},
+
+								"drtm_protection_enabled": {
+									Type:     pluginsdk.TypeBool,
+									Required: true,
+									ForceNew: true,
+								},
+
+								"hvci_protection_enabled": {
+									Type:     pluginsdk.TypeBool,
+									Required: true,
+									ForceNew: true,
+								},
+
+								"side_channel_mitigation_enabled": {
+									Type:     pluginsdk.TypeBool,
+									Required: true,
+									ForceNew: true,
+								},
+
+								"smb_signing_enabled": {
+									Type:     pluginsdk.TypeBool,
+									Required: true,
+									ForceNew: true,
+								},
+
+								"smb_cluster_encryption_enabled": {
+									Type:     pluginsdk.TypeBool,
+									Required: true,
+									ForceNew: true,
+								},
+
+								"wdac_enabled": {
+									Type:     pluginsdk.TypeBool,
+									Required: true,
+									ForceNew: true,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (StackHCIDeploymentSettingResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r StackHCIDeploymentSettingResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 6 * time.Hour,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AzureStackHCI.DeploymentSettings
+
+			var config StackHCIDeploymentSettingModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			stackHCIClusterId, err := deploymentsettings.ParseClusterID(config.StackHCIClusterId)
+			if err != nil {
+				return err
+			}
+			id := deploymentsettings.NewDeploymentSettingID(stackHCIClusterId.SubscriptionId, stackHCIClusterId.ResourceGroupName, stackHCIClusterId.ClusterName, "default")
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			payload := deploymentsettings.DeploymentSetting{
+				Properties: &deploymentsettings.DeploymentSettingsProperties{
+					ArcNodeResourceIds: config.ArcResourceIds,
+					DeploymentMode:     deploymentsettings.DeploymentModeValidate,
+					DeploymentConfiguration: deploymentsettings.DeploymentConfiguration{
+						Version:    pointer.To(config.Version),
+						ScaleUnits: ExpandDeploymentSettingScaleUnits(config.ScaleUnit),
+					},
+				},
+			}
+
+			// the resource exists even validation error
```

Comment:
Calling `metadata.SetID(id)` earlier here because If there is any error in validation and the resource might exist, terraform will mark the resource as [tainted](https://developer.hashicorp.com/terraform/cli/state/taint#the-tainted-status) and remove it in next apply,  so users don't need to manually delete the failed instance or import the resource in next apply. It should work if the resource does not exist, terraform will try to remove it from state during refresh.